versioned/persist/mongodb/src/test/java/org/projectnessie/versioned/persist/mongodb/FlapdoodleMongoTestConnectionProviderSource.java

```diff
@@ -22,82 +22,80 @@
-import de.flapdoodle.embed.process.config.io.ProcessOutput;
-import de.flapdoodle.embed.process.io.StreamProcessor;
 import de.flapdoodle.embed.process.runtime.Network;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import java.io.File;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.concurrent.ThreadLocalRandom;
 
 /**
  * MongoDB test connection-provider source using a locally spawned MongoDB instance via
  * "flapdoodle".
  */
 public class FlapdoodleMongoTestConnectionProviderSource extends MongoTestConnectionProviderSource {
-  private static final Pattern LISTEN_ON_PORT_PATTERN =
-      Pattern.compile(
-          ".*NETWORK ([^\\n]*) waiting for connections on port ([0-9]+)\\n.*",
-          Pattern.MULTILINE | Pattern.DOTALL);
-
-  private final AtomicInteger port = new AtomicInteger();
   private MongodExecutable mongo;
 
+  private File storageDir;
+
   @Override
   public void start() throws Exception {
     if (mongo != null) {
       throw new IllegalStateException("Already started");
     }
 
-    ProcessOutput defaultOutput = MongodProcessOutputConfig.getDefaultInstance(Command.MongoD);
-    StreamProcessor capturedStdout =
-        new StreamProcessor() {
-          private final StringBuilder buffer = new StringBuilder();
-
-          @Override
-          public void process(String block) {
-            if (port.get() == 0) {
-              buffer.append(block);
-              Matcher matcher = LISTEN_ON_PORT_PATTERN.matcher(buffer);
-              if (matcher.matches()) {
-                String portString = matcher.group(2);
-                port.set(Integer.parseInt(portString));
-              }
-            }
-            defaultOutput.output().process(block);
-          }
-
-          @Override
-          public void onProcessed() {
-            defaultOutput.output().onProcessed();
-          }
-        };
-
     MongodStarter starter =
-        MongodStarter.getInstance(
-            Defaults.runtimeConfigFor(Command.MongoD)
-                .processOutput(
-                    ProcessOutput.builder()
-                        .output(capturedStdout)
-                        .error(defaultOutput.error())
-                        .commands(defaultOutput.commands())
-                        .build())
-                .build());
-
-    MongodConfig mongodConfig =
-        MongodConfig.builder()
-            .version(Version.Main.PRODUCTION)
-            .net(new Net(0, Network.localhostIsIPv6()))
-            .build();
-
-    mongo = starter.prepare(mongodConfig);
-    mongo.start();
-
-    assertThat(port).hasPositiveValue();
-
-    String connectionString = String.format("mongodb://localhost:%d", port.get());
+        MongodStarter.getInstance(Defaults.runtimeConfigFor(Command.MongoD).build());
+
+    long timeoutAt = System.currentTimeMillis() + 600_000;
+    int port;
+    while (true) {
```

Comment:
I personally prefer the (current) log scraping approach... it seems cleaner to me and the failure is easy to understand (I hope).

Reply:
Yea, I agree